cli/src/main/java/com/liferay/blade/cli/BladeSettings.java

```diff
@@ -61,12 +61,31 @@ public void load() throws IOException {
 	public void migrateWorkspaceIfNecessary(BladeCLI bladeCLI) throws IOException {
 		WorkspaceProvider workspaceProvider = bladeCLI.getWorkspaceProvider(_settingsFile);
 
-		if (workspaceProvider != null) {
+		if ((workspaceProvider != null) && workspaceProvider.isWorkspace(bladeCLI)) {
 			File workspaceDirectory = workspaceProvider.getWorkspaceDir(_settingsFile);
 
 			File pomFile = new File(workspaceDirectory, "pom.xml");
 
-			if (pomFile.exists() && !_settingsFile.exists()) {
+			boolean shouldPrompt = false;
+
+			if (pomFile.exists()) {
```

Comment:
Is this assuming that only pomFile is all that is needed to make this workspace a maven workspace?  I think we should check the profile provider somehow to make sure it is a maven provider.